vector/src/main/java/im/vector/adapters/NotificationTroubleshootRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,171 @@
+        var fixButton: Button = itemView.findViewById(R.id.button)
+
+
+        fun bind(test: TroubleshootTest) {
+
+            val context = itemView.context
+            titleText.setTextColor(ContextCompat.getColor(context, R.color.riot_primary_background_color_black))
+            descriptionText.setTextColor(ContextCompat.getColor(context, R.color.default_text_light_color_light))
+
+            when (test.status) {
+                TroubleshootTest.TestStatus.NOT_STARTED -> {
+                    titleText.setTextColor(ContextCompat.getColor(context, R.color.primary_text_disabled_material_light))
+                    descriptionText.setTextColor(ContextCompat.getColor(context, R.color.primary_text_disabled_material_light))
+
+                    progressBar.visibility = View.INVISIBLE
+                    statusIconImage.visibility = View.VISIBLE
+                    statusIconImage.setImageResource(R.drawable.abc_btn_check_material)
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                        val color =  ContextCompat.getColor(context, R.color.notification_icon_bg_color)
+                        statusIconImage.imageTintList = ColorStateList.valueOf(color)
+                    }
+                }
+                TroubleshootTest.TestStatus.RUNNING -> {
+                    progressBar.visibility = View.VISIBLE
+                    statusIconImage.visibility = View.INVISIBLE
+
+                }
+                TroubleshootTest.TestStatus.FAILED -> {
+                    progressBar.visibility = View.INVISIBLE
+                    statusIconImage.visibility = View.VISIBLE
+                    statusIconImage.setImageResource(R.drawable.error)
+
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                        statusIconImage.imageTintList = null
+                    }
+
+                    descriptionText.setTextColor(ContextCompat.getColor(context, R.color.error_color_material))
+                }
+                TroubleshootTest.TestStatus.SUCCESS -> {
+                    progressBar.visibility = View.INVISIBLE
+                    statusIconImage.visibility = View.VISIBLE
+                    statusIconImage.setImageResource(R.drawable.abc_btn_check_to_on_mtrl_015)
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                        val color =  ContextCompat.getColor(context, R.color.notification_icon_bg_color)
+                        statusIconImage.imageTintList = ColorStateList.valueOf(color)
+                    }
+
+                }
+            }
+
+            val qf = test.quickFix?.let {
+                // If b is not null.
+                fixButton.text = it.title
+                fixButton.setOnClickListener { _ ->
+                    it.doFix()
+                }
+                fixButton.visibility = View.VISIBLE
+            } ?: run {
```

Comment:
I prefer to use classical if/else pattern instead of ?: run {} which can sometimes behaves not as expected

Reply:
ok